src/update_accumulator.rs

```diff
@@ -1,7 +1,11 @@
 //! TODO
 
 use crate::hardware_data_manager::{HardwareDataManager, Id, Update};
-use std::{cell::RefCell, collections::HashMap, rc::Rc};
+use std::{
+    cell::RefCell,
+    collections::{HashMap, VecDeque},
+    rc::Rc,
+};
 
 /// The `UpdateAccumulator` consumes updates from a `HardwareDataManager`, and
 /// accumulates them. It can be queried for the most recent updates using `.get_status()`.
```

Comment:
This is excellent! To incorporate this into the pipeline, I think I can wrap the `UpdateAccumulator` in a `Component` thread, have it not read from its input channel. Every call to `convert()` can instead invoke `get_status()` and send the resulting `Update`s it gets back to its output channel.

If I were to write my part over again in the future, I might have opted for shared data queues vs using channels to pass data. This looks great!